go/coredns/plugin/pfdns/pfdns.go

```diff
@@ -285,10 +285,23 @@ func (pf *pfdns) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 	switch state.Family() {
 	case 1:
 		rr = new(dns.A)
-		if state.QName() == pf.PortalFQDN {
-			rr.(*dns.A).Hdr = dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass(), Ttl: 60}
-		} else {
-			rr.(*dns.A).Hdr = dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass(), Ttl: 15}
+		var found bool
+		found = false
+		for i := 0; i <= len(pf.PortalFQDN); i++ {
```

Comment:
This should be an array sorted by network length to avoid matching against the larger network first.